### DISCUSSION TITLE
Converting a generic html/css template for Yellow?

### DISCUSSION THREAD

**Discussion by tyeeman** [datenstrom/community#366] [Ask a question] [answered]
Been testing Yellow, it just works great! Great Documentation also!
I like to choose my own templates so I'm wondering how to convert a generic html template to Yellow without spending hours doing so. It looks like snippets are the key. Do I take my generic template and make it into snippets by replacing lots of code with the Yellow php codes like in content-blog.php? There are lots of divs in there with corresponding styling so I'm not sure if I just leave content-blog.php alone for my conversion? I didn't find anything in the docs on doing something like this but obviously you had to when you created these templates for Yellow in the first place.

**Comment by tyeeman** [comment, marked as answer]
Looks like I just have to copy and paste the desired php code into my template where ever they should go and that's it.

**Comment by markseuffert** [comment]
Look like you found out how it works.

Want to give us some feedback about making a website with Datenstrom Yellow? I pushed back some  ideas and backwards-compatibility-breaking changes, so the next release will probably bring some bigger changes. Here are two examples: Markdown files with `md` extenion (which is more common), a different folder structure for templates (which could be called layouts in the future).

What do you think?

**Comment by tyeeman** [comment]
I looked at Yellow over a year ago and dismissed it because at that time I thought that to convert a generic template looked way too hard, but now I don't think that so that's why I came back to Yellow. In the last few months I have been looking at static generators and my current favourite does limit the number of blog posts on the front page. All are shown no matter how many, so luckily I found out Yellow can generate a static site and define the number of default posts on the front page (please confirm that BlogPaginationLimit: 5 is what does that?), and it has tags which to me is very important. I also changed the default extension to .md in the config file a couple of days ago. So there you go. I will try to convert a generic template sometime this week. The only thing that I would like is to add a slideshow play button to the gallery with a dissolve transition to make it look a little more professional. That could probably be done with javascript but I'm not a coder but I have seen galleries using javascript on static sites.

And regarding Netlify, you said it would be very hard to add .html to the end of pages or posts in the urls. I guess you do it differently than other php scripts. I thought you could just add some code to make it work but I guess not. I may just have to add to that netlify.toml file any folder I want rendered as text/html, which is not hard at all to do. 

Sometimes I get confused with the words templates and layouts. I'm used to taking a generic template as I like to call it and just add php tags into it where ever they are needed, like I'm going to do this week with Yellow. So when I hear of layouts also, I get confused until I actually look into what the developer is using them for. Personally I only need the generic template that is actually the layout also.

**Comment by markseuffert** [comment]
I can confirm that `BlogPaginationLimit` is the number of pages shown on the blog. 

When it comes to templates you have full control over HTML, CSS and JS. You can see Datenstrom Yellow as the glue code, that puts things together. There are no dependencies, no big frameworks. Developers should be able to add features, so that they get the results they want. There are image gallery plugins for popular JavaScript libraries. If your favourite JavaScript library is missing, consider writing a new plugin.

About Netlify, the workaround you found is the best solution.

**Comment by tyeeman** [comment]
So is ```BlogPaginationLimit``` the number of posts per page? If not, what limits the number per page?

**Comment by markseuffert** [comment]
Yes you're correct. The pagination limits how many are shown at once, breaks it up into multiple pages.
